v03_pipeline/lib/tasks/reference_data/updated_reference_dataset_collection.py

```diff
@@ -22,6 +33,56 @@ def _destination_path(self) -> str:
             self.reference_dataset_collection,
         )
 
+    def complete(self) -> bool:
+        self._datasets_to_update.clear()
+
+        if not self.output().exists():
+            logger.info(
+                f'Creating a new reference dataset collection for {self.reference_genome}, {self.dataset_type}, '
+                f'{self.reference_dataset_collection} because it does not exist',
+            )
+            self._datasets_to_update.extend(
+                self.reference_dataset_collection.datasets(self.dataset_type),
+            )
+            self._should_validate = False
+            return False
+
+        if not self._should_validate:
+            return super().complete()
+
+        logger.info(
+            f'Reading existing reference dataset collection from {self._destination_path}',
+        )
+        joined_ht = hl.read_table(self._destination_path)
+        for dataset in (
+            [self.dataset]
+            if self.dataset is not None
+            else self.reference_dataset_collection.datasets(self.dataset_type)
+        ):
+            if dataset not in joined_ht.row:
```

Comment:
I think it'd be cleaner to push this logic, and the `for dataset in 'datasets'` loop into the `validate_joined_ht_globals_match_config`.  To me, the "dataset present in one but not in the other" case should be treated as a validation step similar to the others.  It also makes unit testing this case easier as it no longer is bundled into a luigi method.

Reply:
good call, I agree